operator-common/src/main/java/io/strimzi/operator/common/operator/resource/CrdOperator.java

```diff
@@ -30,13 +42,50 @@
      */
     public CrdOperator(Vertx vertx, C client, Class<T> cls, Class<L> listCls, Class<D> doneableCls) {
         super(vertx, client, Crds.kind(cls));
+        this.baseUrl = this.client.getMasterUrl().toString();
         this.cls = cls;
         this.listCls = listCls;
         this.doneableCls = doneableCls;
+
+        if (cls.equals(Kafka.class)) {
+            this.plural = Kafka.RESOURCE_PLURAL;
+        } else {
+            this.plural = null;
+        }
     }
 
     @Override
     protected MixedOperation<T, L, D, Resource<T, D>> operation() {
         return Crds.operation(client, cls, listCls, doneableCls);
     }
+
+    public void updateStatus(T resource) {
+        try {
+            OkHttpClient client = this.client.adapt(OkHttpClient.class);
+            RequestBody postBody = RequestBody.create(OperationSupport.JSON, new ObjectMapper().writeValueAsString(resource));
+
+            Request request = new Request.Builder().put(postBody).url(
+                    this.baseUrl + "apis/" + resource.getApiVersion() + "/namespaces/" + resource.getMetadata().getNamespace()
+                    + "/" + this.plural + "/" + resource.getMetadata().getName() + "/status").build();
```

Comment:
Do we need any new RBAC rights to set the status subresources?

Reply:
Yes, I had add this line to get access rights: https://github.com/strimzi/strimzi-kafka-operator/blob/892aef2da81ed2b8229a36e93e69532d296916ad/install/cluster-operator/020-ClusterRole-strimzi-cluster-operator-role.yaml#L44